woodchipper/http/fastapi.py

```diff
@@ -0,0 +1,73 @@
+import uuid
+from typing import Any, Callable, Coroutine
+
+from fastapi import FastAPI, Request, Response
+from fastapi.routing import APIRoute
+
+from woodchipper.context import LoggingContext, logging_ctx
+
+BLACKLISTED_HEADERS = ["authorization", "cookie"]
+
+
+class WoodchipperFastAPI:
+    def __init__(self, app: FastAPI, blacklisted_headers=BLACKLISTED_HEADERS, request_id_factory=None):
+        self._app = app
+        self._blacklisted_headers = blacklisted_headers
+        self._request_id_factory = request_id_factory or (lambda: str(uuid.uuid4()))
+
+    def get_custom_route_handler(self, gen_id, blacklisted_headers):
+        def get_route_handler(self) -> Callable[[Request], Coroutine[Any, Any, Response]]:
+
+            original_route_handler = super(type(self), self).get_route_handler()
+
+            async def woodchipper_route_handler(request: Request) -> Response:
```

Comment:
When testing this, if a route does not have a matching handler (404 situation) it doesn't get this information added to it.

Thoughts on how to get around that? If you do find a way around that, could you please also add a test to confirm?

Reply:
I looked into this and 404s are thrown in the routing middleware which runs _before_ our code is even called. If we wanted to catch 404s we would need to switch to doing the context creation in our own ASGI middleware. This is very possible but does increase the complexity. I'll refrain from continued work on this PR until this issue is decided as it would make most of the rest of the work moot.

Reply:
Nah, that type of investigation was what I was looking for -- a reason for why it would be difficult to do here.

Not having that is fine with me. It will mean that we miss parity with Flask on release of this, but I can open an issue ticket around the missing part.

I'm using FastAPI too so I may work on it. Would be a good reason to dig more into FastAPI.